crates/settings/src/network.rs

```diff
@@ -27,18 +37,126 @@ pub struct Network {
 impl Network {
     pub fn dummy() -> Self {
         Network {
-            name: "".to_string(),
+            document: Arc::new(RwLock::new(StrictYaml::String("".to_string()))),
+            key: "".to_string(),
             rpc: Url::parse("http://rpc.com").unwrap(),
             chain_id: 1,
             label: None,
             network_id: None,
             currency: None,
         }
     }
+
+    pub fn update_rpc(&mut self, rpc: &str) -> Result<(), YamlError> {
+        let mut document = self
+            .document
+            .write()
+            .map_err(|_| YamlError::WriteLockError)?;
+
+        if let StrictYaml::Hash(ref mut document_hash) = *document {
+            if let Some(StrictYaml::Hash(ref mut networks)) =
+                document_hash.get_mut(&StrictYaml::String("networks".to_string()))
+            {
+                if let Some(StrictYaml::Hash(ref mut network)) =
+                    networks.get_mut(&StrictYaml::String(self.key.to_string()))
+                {
+                    network[&StrictYaml::String("rpc".to_string())] =
+                        StrictYaml::String(rpc.to_string());
+                    self.rpc =
+                        Url::parse(rpc).map_err(ParseNetworkConfigSourceError::RpcParseError)?;
```

Comment:
could this validation be split out so it doesn't need to be repeated across the setters and getters?